cmd/juju/commands/bootstrap_clouds.go

```diff
@@ -133,3 +138,105 @@ func printCloudRegions(ctx *cmd.Context, cloudName string) error {
 	}
 	return nil
 }
+
+func (c *bootstrapCommand) printCloudConfig(ctx *cmd.Context, showCloudConfigInput string) error {
+	// Print the general cloud config attributes.
+	var env environschema.Fields
+	schema, _ := config.Schema(env)
+	defaults := config.ConfigDefaults()
+	values, err := configMapToPrint(schema, defaults)
+	if err != nil {
+		return err
+	}
+
+	// Print the cloud config attributes specific to its provider.
+	if showCloudConfigInput != "" {
+		var cloudType string
+		if cloud, err := common.CloudByName(showCloudConfigInput); err == nil {
+			cloudType = cloud.Type
+		} else {
+			cloudType = showCloudConfigInput
+		}
+		provider, err := environs.Provider(cloudType)
```

Comment:
This logic to get the provider config should be factored out to a shared function used by both show-cloud and bootstrap